CMakeLists.txt

```diff
@@ -664,6 +664,13 @@ elseif(MSVC)
 	add_compile_options("/utf-8")
 ENDIF()
 
+# gcc builds support gprof for profiling, should not be used for release builds because
+# -fomit-frame-pointer clashes with -pg
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT CMAKE_BUILD_TYPE STREQUAL "Release")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pg")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pg")
```

Comment:
Modifying `CMAKE_<LANG>_FLAGS` is a rather outdated approach to adding compiler options. In this case, the alternative to use is `add_compile_options`. Also, according to [the documentation for `-pg`](https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Instrumentation-Options.html#index-pg), this flag needs to be passed to the linker too, so you will also need to use `add_link_options`.

Reply:
also done.